mmv1/third_party/terraform/tpgresource/regional_utils.go

```diff
@@ -20,9 +20,9 @@ func IsZone(location string) bool {
 // - zone argument set in the provider config
 func GetLocation(d TerraformResourceData, config *transport_tpg.Config) (string, error) {
 	if v, ok := d.GetOk("location"); ok {
-		return v.(string), nil
+		return GetResourceNameFromSelfLink(v.(string)), nil
 	} else if v, isRegionalCluster := d.GetOk("region"); isRegionalCluster {
-		return v.(string), nil
+		return GetResourceNameFromSelfLink(v.(string)), nil
 	} else {
 		// If region is not explicitly set, use "zone" (or fall back to the provider-level zone).
```

Comment:
(Relates to the other comment about the confusing test case you had commented out)

[This test case](https://github.com/hashicorp/terraform-provider-google/blob/643f396836a197aa8cd4605bfed77951039e8f62/google/tpgresource/utils_test.go#L316-L321) should be made to fail by updating this function to also search for locations in this order:

1. location on resource, if available
2. if not, region on resource if available
3. if not, zone on resource if available 
4. if not, region on provider if available
5. if not, zone on provider if available
6. if none of the fields above are set return an error

Currently the function does 1, 2, 3+5+6* (Those three all done by `GetZone`)

You'd need to update the function so that it includes 4 after 3, which will involve removing using of `GetZone`.


(*) Also, I think the error message returned from this function needs to be changed. Take a look at the error returned from this function and then imagine you were a user experiencing that error when provisioning a resource that has a `location` field. Could you come up with a better error message for this function to return instead?